agents/agents/executor/types/db_state.go

```diff
@@ -20,6 +20,8 @@ type DBState struct {
 	Proof *json.RawMessage
 	// StateIndex is the index of the state in the Snapshot.
 	StateIndex *uint32
+	// BlockNumber is the block number the state was received at on Summit.
+	BlockNumber *uint64
```

Comment:
I don't really want to address in this pr, as I think it's big enough, but I'm a really big fan of hiding this kind of db implementation stuff from the rest of the program since that can get messy

Reply:
Added TODO's for each db_<type>.go

Reply:
if this is just a mask, let's move it to the db package. No circular dependency issues tehre